atlassian_jwt_auth/signer.py

```diff
@@ -72,11 +72,14 @@ def generate_jwt(self, audience, **kwargs):
             self.issuer)
         private_key = self._obtain_private_key(
             key_identifier, private_key_pem)
-        return jwt.encode(
+        token = jwt.encode(
             self._generate_claims(audience, **kwargs),
             key=private_key,
             algorithm=self.algorithm,
             headers={'kid': key_identifier.key_id})
+        if isinstance(token, str):
+            token = token.encode('utf-8')
+        return token
```

Comment:
To remain consistent.

Reply:
Ah really, PyJWT changed between bytes and str between versions?

OOC, if and when support only PyJWT 2+, will we also change this to be in line with the types it returns?

Reply:
Possibly, although it might result in un-needed churn :shrug: .